internal/impl/amqp1/input.go

```diff
@@ -488,6 +494,8 @@ func amqpSetMetadata(p *service.Message, k string, v any) {
 		metaValue = ""
 	case string:
 		metaValue = v
+	case *string:
+		metaValue = *v
```

Comment:
Yes, of course this would work - and in my first approach I had it like that. 

The reason I've made it bit more complex (and generic) is, because the `MessageProperties` has even more pointer fields like `*time.Time` and `*uint32` and I wanted to avoid lots of `if's`. Also the amqp message itself has way more additional properties like `Header` , `ApplicationProperties` etc. which could be of interesst (or added in future PR's 😏 ). 

I feel like that the most part of `func amqpSetMetadata` is a generic `any to string` conversion anyways. This is even a bit redundant with the same function of the amqp09 component and probably something that can be generalized outside of the specific components. But I'm not sure if it is intended like that or just "lazy copy & paste".